clang_tools/util.py

```diff
@@ -99,3 +99,21 @@ def verify_sha512(checksum: str, exe: bytes) -> bool:
         # released checksum's include the corresponding filename (which we don't need)
         checksum = checksum.split(" ", 1)[0]
     return checksum == hashlib.sha512(exe).hexdigest()
+
+
+def parse_version(version: str) -> Tuple[int]:
+    """Parse the given version string into a semantic specification.
+
+    :param version: The version specification as a string.
+
+    :returns: A tuple of ints that describes the major, minor, and patch versions.
+        If the version is a path, then the tuple is just 3 zeros.
+    """
+    version_tuple = version.split(".")
+    if len(version_tuple) < 3:
+        # append minor and patch version numbers if not specified
+        version_tuple += ["0"] * (3 - len(version_tuple))
+    try:
+        return tuple([int(x) for x in version_tuple])
+    except ValueError:
+        return (0, 0, 0)
```

Comment:
Here, it is assumed that the version is a path (& just returns `(0, 0, 0)`). However, it might be better to assert that the path exists also.

Reply:
done.

I also ran mypy on the pkg/tests sources, just to be sure.